internal/commands/pull/inputs.go

```diff
@@ -109,3 +117,62 @@ func (i *inputs) resolveRemoteApp(ui terminal.UI, clients cli.Clients) (realm.Ap
 
 	return app, nil
 }
+
+func (i *inputs) resolveClient(ui terminal.UI, realmClient realm.Client, groupID, appID string) (map[string]*zip.Reader, error) {
+	compatibleTemplates, ok, err := realmClient.CompatibleTemplates(groupID, appID)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, nil
+	}
+
+	if i.TemplateID != "" {
+		for _, template := range compatibleTemplates {
+			// This is the ID that they selected
+			if template.ID == i.TemplateID {
+				// Fetch the template
+				templateZip, err := realmClient.ClientTemplate(groupID, appID, template.ID)
+				if err != nil {
+					return nil, err
+				}
+				return map[string]*zip.Reader{template.ID: templateZip}, nil
+			}
+		}
+		return nil, fmt.Errorf("template %s is not compatible with this app", i.TemplateID)
+	}
+
+	if len(compatibleTemplates) != 0 {
+		if proceed, err := ui.Confirm("Would you like to export with a template?"); err != nil {
+			return nil, err
+		} else if proceed {
+			nameOptions := make([]string, len(compatibleTemplates))
+			for idx, compatibleTemplate := range compatibleTemplates {
+				nameOptions[idx] = fmt.Sprintf("[%s]: %s", compatibleTemplate.ID, compatibleTemplate.Name)
+			}
+
+			var selectedTemplateIdxs []int
+			if err := ui.AskOne(
+				&selectedTemplateIdxs,
+				&survey.MultiSelect{Message: "Which template(s) would you like to export this app with", Options: nameOptions}); err != nil {
+				return nil, err
+			}
+
+			templateIDs := make([]string, len(selectedTemplateIdxs))
+			for idx, selectedTemplateIdx := range selectedTemplateIdxs {
+				templateIDs[idx] = compatibleTemplates[selectedTemplateIdx].ID
+			}
+
+			result := make(map[string]*zip.Reader, len(templateIDs))
+			for _, templateID := range templateIDs {
+				templateZip, err := realmClient.ClientTemplate(groupID, appID, templateID)
```

Comment:
this is the client method where we're returning `nil, nil`, and so now we're populating a map entry with a `nil` `*zip.Reader`.  make sure this is ok with the consuming code for this method (where we're writing the frontend templates to disk).

and if we did add the `bool` return here, then it would simply be a `if !ok { continue }` (and it would seem you'd also want to warn the user about such a situation somehow ... if that's the case please try to keep all the warning/debug messages printed in the command `Handler` code and not here in the inputs ... also make sure to point that out in the `dryRun` stuff too)

sorry for the run-on comment there, let me know if i can clarify anything there